warp10/src/main/java/io/warp10/continuum/store/SlicedRowFilterGTSDecoderIterator.java

```diff
@@ -87,30 +101,42 @@
    */
   private Result pendingresult = null;
   
-  private static final byte[] ZERO_BYTES = Longs.toByteArray(0L);
-  private static final byte[] ONES_BYTES = Longs.toByteArray(0xffffffffffffffffL);
-  
   private static byte[] prefix = Constants.HBASE_RAW_DATA_KEY_PREFIX;
 
   private final boolean writeTimestamp;
+  private final boolean hasStep;
+  private final boolean hasTimestep;
+  private final boolean hasSample;
+  private long skip = 0L;
+  private long steps = 0L;
+  private final long step;
+  private final long timestep;
+  private long nextTimestamp = Long.MAX_VALUE;
+  private final double sample;
   
-  public SlicedRowFilterGTSDecoderIterator(long now, long timespan, List<Metadata> metadatas, Connection conn, TableName tableName, byte[] colfam, boolean writeTimestamp, KeyStore keystore, boolean useBlockCache) {
+  private Random prng = null;
+
+  public SlicedRowFilterGTSDecoderIterator(FetchRequest req, Connection conn, TableName tableName, byte[] colfam, KeyStore keystore, boolean useBlockCache) {
       
+    this.request = req;
     this.keystore = keystore;
-    this.now = now;
-    this.timespan = timespan;
+    this.now = req.getNow();
+    this.then = req.getThents();
+    this.count = req.getCount();
+    this.sample = req.getSample();
+    this.hasSample = this.sample < 1.0D;
+    this.prng = hasSample ? new Random() : null;
+    this.hasStep = req.getStep() > 1L;
+    this.steps = req.getStep() - 1L;
```

Comment:
I forgot this one, initialize to 0L to avoid stepping over the first values, even if it's not used yet.